### PR TITLE
Make sure latest data is always sent to callback when using cached data

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett PTAL

cc @liggitt

File: app/scripts/services/data.js [hunk 1]
```diff
@@ -334,27 +334,41 @@ angular.module('openshiftConsole')
...
-        }, 0);
-      }
-      else {
-        $timeout(function() {
+      $timeout(function() {
+        // re-pull the data from the cache in case it has changed on us from the watch
+        var data = this._data(key);
+        if (!data) {
+          // Extremely unlikely, would require that the watch was closed and other watches pushed
+          // this data out of the cache
```
Comment (jwforres) [inline app/scripts/services/data.js]: the alternative in this scenario is to go do the GET if we've lost the entire cached data map in the interim, but it means a significantly bigger refactor, if are going to cherry-pick this into 1.3.1 i'd feel better with a less risky refactor

File: app/scripts/services/data.js [hunk 2]
```diff
@@ -578,9 +592,13 @@ DataService.prototype.createStream = function(resource, name, context, opts, isR
     else {
       if (callback) {
         var existingData = this._data(key);
+        var resourceVersion = this._resourceVersion(key);
         if (existingData) {
```
Comment (spadgett) [inline app/scripts/services/data.js]: I might even remove the existingData var 

```js
if (this._data(key)) ...
```

so no one changes it back?

**Comment by jwforres** [comment]
updated

**Comment by spadgett** [comment]
LGTM thanks

**Comment by spadgett** [comment]
[merge]

**Pull request merged**